src/Microsoft.Health.Fhir.TemplateManagement/Overlay/OverlayOperator.cs

```diff
@@ -0,0 +1,206 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Health.Fhir.Liquid.Converter.Models;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+using Microsoft.Health.Fhir.TemplateManagement.Utilities;
+using Newtonsoft.Json;
+using SharpCompress.Common;
+using SharpCompress.Writers;
+using SharpCompress.Writers.Tar;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Overlay
+{
+    public class OverlayOperator : IOverlayOperator
+    {
+        public OCIFileLayer ExtractOCIFileLayer(OCIArtifactLayer artifactLayer)
+        {
+            if (artifactLayer == null)
+            {
+                return null;
+            }
+
+            var artifacts = StreamUtility.DecompressTarGzStream(new MemoryStream(artifactLayer.Content));
+            var metaBytes = artifacts.GetValueOrDefault(Constants.OverlayMetadataFile);
+            OverlayMetadata metaJson = null;
+            if (metaBytes != null)
+            {
+                try
+                {
+                    metaJson = JsonConvert.DeserializeObject<OverlayMetadata>(Encoding.UTF8.GetString(metaBytes));
+                }
+                catch (Exception ex)
+                {
+                    throw new OverlayException(TemplateManagementErrorCode.OverlayMetaJsonNotFound, "overlayMeta.json file invalid", ex);
+                }
+            }
+
+            artifacts.Remove(Constants.OverlayMetadataFile);
+            return new OCIFileLayer() { Content = artifactLayer.Content, FileContent = artifacts, FileName = artifactLayer.FileName, SequenceNumber = metaJson == null ? -1 : metaJson.SequenceNumber };
+        }
+
+        public List<OCIFileLayer> ExtractOCIFileLayers(List<OCIArtifactLayer> artifactLayers)
+        {
```

Comment:
We can use a single line `return artifactLayers?.Select(ExtractOCIFileLayer).ToList();` as the function's body to reduce several lines here.